File: ruby/math/gcd_lcm.rb
#!/usr/bin/env ruby
# Print the divisor and multiple of fixed pairs, with Euclid's algorithm written out rather than taken from Integer.

PAIRS = [[1071, 462], [270, 192], [17, 5], [120, 36]].freeze

def gcd(first, second)
  first, second = second, first % second until second.zero?
  first
end

PAIRS.each do |first, second|
  divisor = gcd(first, second)
  puts "#{first} #{second} #{divisor} #{first / divisor * second}"
end

File: php/math/gcd_lcm.php
<?php
// Print the divisor and multiple of fixed pairs, with Euclid's algorithm written as a loop.

declare(strict_types=1);

const PAIRS = [[1071, 462], [270, 192], [17, 5], [120, 36]];

function gcd(int $first, int $second): int
{
    while ($second !== 0) {
        [$first, $second] = [$second, $first % $second];
    }

    return $first;
}

foreach (PAIRS as [$first, $second]) {
    $divisor = gcd($first, $second);
    printf("%d %d %d %d\n", $first, $second, $divisor, intdiv($first, $divisor) * $second);
}
